### PR TITLE
DEP: Enforce respecting group_keys in groupby.apply

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/groupby/test_apply.py [hunk 1]
```diff
@@ -1011,25 +1011,6 @@ def test_result_order_group_keys_false():
     tm.assert_frame_equal(result, expected)
 
 
-def test_groupby_apply_group_keys_warns():
-    df = DataFrame({"A": [0, 1, 1], "B": [1, 2, 3]})
-    msg = "Not prepending group keys to the result index"
-    with tm.assert_produces_warning(FutureWarning, match=msg):
-        result = df.groupby("A").apply(lambda x: x)
```
Comment (mroeschke) [inline pandas/tests/groupby/test_apply.py:1018]: Is it worth keeping these tests with `group_keys=False` for testing coverage, or would that be redundant with existing tests?

Comment (rhshadrach) [reply pandas/tests/groupby/test_apply.py:1018]: As part of the original deprecation, many tests that use apply now have `group_keys=False`. Throughout the tests, there are 88 lines of code where we specify `group_keys=False`. I feel pretty good about our test coverage without this, but no real objection if you want it kept.

Comment (mroeschke) [reply pandas/tests/groupby/test_apply.py:1018]: Sounds good. I'm fine with removing these tests then.

**Comment by mroeschke** [comment]
Merge conflict otherwise LGTM

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**